lilia/libraries/thirdparty/tween.lua

```diff
@@ -0,0 +1,554 @@
+TWEEN_EASE_QUINT_OUT = math.ease.OutQuint
+TWEEN_EASE_QUINT_IN_OUT = math.ease.InOutQuint
+TWEEN_EASE_EXPO_IN = math.ease.InExpo
+TWEEN_EASE_EXPO_OUT = math.ease.OutExpo
+TWEEN_EASE_EXPO_IN_OUT = math.ease.InOutExpo
+TWEEN_EASE_CIRC_IN = math.ease.InCirc
+TWEEN_EASE_CIRC_OUT = math.ease.OutCirc
+TWEEN_EASE_CIRC_IN_OUT = math.ease.InOutCirc
+TWEEN_EASE_BACK_IN = math.ease.InBack
+TWEEN_EASE_BACK_OUT = math.ease.OutBack
+TWEEN_EASE_BACK_IN_OUT = math.ease.InOutBack
+TWEEN_EASE_ELASTIC_IN = math.ease.InElastic
+TWEEN_EASE_ELASTIC_OUT = math.ease.OutElastic
+TWEEN_EASE_ELASTIC_IN_OUT = math.ease.InOutElastic
+TWEEN_EASE_BOUNCE_IN = math.ease.InBounce
+TWEEN_EASE_BOUNCE_OUT = math.ease.OutBounce
+TWEEN_EASE_BOUNCE_IN_OUT = math.ease.InOutBounce
+
+local function table_Inherit(target, base)
+	for k, v in next, base do
+		if target[k] then continue end
+		
+		target[k] = v
+	end
+	
+	return target
+end
+
+local metaTable_Vector2 = {
+	__add = function(self, other)
+		return Vector2(self.x + other.x, self.y + other.y)
+	end,
+	
+	__sub = function(self, other)
+		return Vector2(self.x - other.x, self.y - other.y)
+	end,
+	
+	__mul = function(self, other)
+		if isnumber(other) then
+			return Vector2(self.x * other, self.y * other)
+		elseif isnumber(self) then
+			return Vector2(self * other.x, self * other.y)
+		end
+		
+		return Vector2(self.x * other.x, self.y * other.y)
+	end,
+	
+	__div = function(self, other)
+		if isnumber(other) then
+			return Vector2(self.x / other, self.y / other)
+		end
+		
+		return Vector2(self.x / other.x, self.y / other.y)
+	end,
+	
+	__tostring = function(self)
+		return string.format("%.6f %.6f", self.x, self.y)
+	end,
+	
+	SetUnpacked = function(self, x, y)
+		self.x = x
+		self.y = y
+	end,
+	
+	GetPos = function(self)
+		return Vector2(self.x, self.y)
+	end
+}
+
+metaTable_Vector2.__index = metaTable_Vector2
+
+function isvector2(v)
+	return getmetatable(v) == metaTable_Vector2
+end
+
+function Vector2(x, y)
+	local Vector2 = {
+		x = x,
+		y = y
+	}
+	
+	return setmetatable(Vector2, metaTable_Vector2)
+end
+
+function tween.Lerp(from, to, t)
+	return (1 - t) * from + t * to
+end
+
+local Lerp = tween.Lerp
+
+local function CalcAng(from, to)
+	local diff = math_fmod(to - from, 360)
+	
+	return math_fmod(2 * diff, 360) - diff
+end
+
+local function LerpDeg(from, to, t)
+	return from + CalcAng(from, to) * t
+end
+
+function tween.LerpAngle(from, to, t)
+	return Angle(
+		LerpDeg(from.p, to.p, t),
+		LerpDeg(from.y, to.y, t),
+		LerpDeg(from.r, to.r, t)
+	)
+end
+
+local LerpAngle = tween.LerpAngle
+
+local function LerpVector2Unpacked(vector2, from, to, t)
+	local lerped_vector2 = Lerp(from, to, t)
+	
+	vector2:SetUnpacked(lerped_vector2.x, lerped_vector2.y)
+end
+
+local function LerpVectorUnpacked(vector, from, to, t)
+	local lerped_vector = Lerp(from, to, t)
+	
+	vector:SetUnpacked(lerped_vector.x, lerped_vector.y, lerped_vector.z)
+end
+
+local function LerpColor(from, to, t)
```

Comment:
I'd suggest the following to not be local functions, but rather global

Reply:
Done I had made it local on accident when typing